partner_firstname/tests/test_copy.py

```diff
@@ -0,0 +1,57 @@
+# -*- coding: utf-8 -*-
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+"""Test copy function for models."""
+
+from odoo.tests.common import TransactionCase
+from .base import MailInstalled
+
+
+class UserCase(TransactionCase, MailInstalled):
+    """Test ``res.users``."""
+    model = "res.users"
+    context = {"default_login": "user@example.com"}
+
+    def setUp(self):
+        super(UserCase, self).setUp()
+        self.create_original()
+
+    def create_original(self):
+        self.original = self.env["res.users"].create({
+            "firstname": u"Firstname",
+            "lastname": u"Lastname",
+            "name": u"Firstname Lastname",
+            "login": u"firstname.lastname"
+        })
+
+    def tearDown(self):
+        super(UserCase, self).tearDown()
+
+    def compare(self, copy):
+        self.assertEqual(copy.lastname, u"Lastname2")
+        self.assertEqual(copy.firstname, u"Firstname2")
+        self.assertEqual(copy.name, u"Lastname2 Firstname2")
+
+    def test_copy_name(self):
+        """Copy original with default name set - firstname lastname not set."""
+        copy = self.original.copy({
```

Comment:
In your copy methods use the default= keyword. It works without, but it is clearer to call a keyword argument with a keyword.

Reply:
done